app/src/main/java/io/gnosis/safe/utils/DateUtils.kt

```diff
@@ -0,0 +1,12 @@
+package io.gnosis.safe.utils
+
+import org.threeten.bp.Instant
+import org.threeten.bp.ZoneId
+import org.threeten.bp.format.DateTimeFormatter
+import org.threeten.bp.format.FormatStyle
+import java.util.*
+
+fun Date.formatBackendDate(zoneId : ZoneId = ZoneId.systemDefault(), locale: Locale = Locale.getDefault()): String {
+    val customFormatter: DateTimeFormatter = DateTimeFormatter.ofLocalizedDateTime(FormatStyle.MEDIUM).withZone(zoneId).withLocale(locale)
```

Comment:
This is not a performance problem. I checked that the `customFormatter` is cached. The first call takes between 16ms and 40ms. All subsequent calls take between 0ms and 1ms.